packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -80,6 +81,86 @@ export function isTreeNodeSchemaClass<
  */
 export type AllowedTypes = readonly LazyItem<TreeNodeSchema>[];
 
+/**
+ * Stores annotations for a set of allowed types.
+ * @alpha
+ */
+export interface AnnotatedAllowedTypes {
+	/**
+	 * Annotations that apply to a set of allowed types.
+	 */
+	readonly metadata: AllowedTypesMetadata;
+	/**
+	 * All the allowed types that the annotations apply to. The types themselves may also have individual annotations.
+	 */
+	readonly types: readonly (AnnotatedAllowedType | LazyItem<TreeNodeSchema>)[];
+}
+
+/**
+ * Checks if the input is an {@link AnnotatedAllowedTypes}.
+ */
+export function isAnnotatedAllowedTypes(
+	allowedTypes: ImplicitAnnotatedAllowedTypes,
+): allowedTypes is AnnotatedAllowedTypes {
+	return (
+		typeof allowedTypes === "object" && "metadata" in allowedTypes && "types" in allowedTypes
+	);
+}
+
+/**
+ * Annotations that apply to a set of allowed types.
+ * @remarks
+ * Additional optionals may be added to this as non-breaking changes, so implementations of it should be simple object literals with no unlisted members.
+ * @alpha
+ */
+export interface AllowedTypesMetadata {
+	/**
+	 * User defined metadata
+	 */
+	custom?: unknown;
+}
+
+/**
+ * Stores annotations for an individual allowed type.
+ * @alpha
+ */
+export interface AnnotatedAllowedType<
+	T extends LazyItem<TreeNodeSchema> = LazyItem<TreeNodeSchema>,
+> {
+	/**
+	 * Annotations for the allowed type.
+	 */
+	readonly metadata: AllowedTypeMetadata;
+	/**
+	 * The allowed type the annotations apply to in a particular schema.
+	 */
+	readonly type: T;
```

Comment:
I think it would make more sense to parameterize this over the schema than the reference to the schema.

```suggestion
export interface AnnotatedAllowedType<
	T extends TreeNodeSchema = TreeNodeSchema,
> {
	/**
	 * Annotations for the allowed type.
	 */
	readonly metadata: AllowedTypeMetadata;
	/**
	 * The allowed type the annotations apply to in a particular schema.
	 */
	readonly type: LazyItem<T>;
```